internal/provider/data_source_public_key_test.go

```diff
@@ -43,34 +50,100 @@ func TestAccPublicKey_dataSource(t *testing.T) {
 				`,
 				Check: resource.TestCheckResourceAttrPair(
 					"data.tls_public_key.test", "public_key_pem",
-					"tls_private_key.test", "public_key_pem"),
+					"tls_private_key.test", "public_key_pem",
+				),
 			},
 			{
 				Config: `
-					resource "tls_private_key" "key" {
+					resource "tls_private_key" "ecdsaPrvKey" {
 						algorithm   = "ECDSA"
 						ecdsa_curve = "P384"
 					}
-					data "tls_public_key" "pub" {
-						private_key_pem = "${tls_private_key.key.private_key_pem}"
+					data "tls_public_key" "ecdsaPubKey" {
+						private_key_pem = "${tls_private_key.ecdsaPrvKey.private_key_pem}"
```

Comment:
Nit: Terraform 0.12+ syntax can/should be used in the test code since it is generally easier to read. Also if there is just a singular resource / data source of a type, it is generally easiest to name them `test` so you don't need to make up naming (similar to `example` in documentation examples), but that's its own project.

```suggestion
						private_key_pem = tls_private_key.ecdsaPrvKey.private_key_pem
```

Aside: We could theoretically go down the [`terrafmt`](https://github.com/katbyte/terrafmt) road for these providers to enforce these types of Terraform configuration issues via CI tooling, but it is probably more worth spending time on the new testing framework and documentation generation instead.